### PR TITLE
refactor(sidenav): split into drawer and sidenav

### PR DESCRIPTION
**Pull request by mmalerba**

### CODE REVIEW

File: src/demo-app/sidenav/sidenav-demo.html [hunk 1]
```diff
@@ -15,7 +15,7 @@
     <input #myinput>
   </md-sidenav>
 
-  <md-sidenav #end align="end">
+  <md-sidenav #end side="end">
```
Comment (jelbourn) [inline src/demo-app/sidenav/sidenav-demo.html]: What do you think of calling this `position`?

Comment (mmalerba) [inline src/demo-app/sidenav/sidenav-demo.html]: That sounds good to me, I just didn't like `align`, sounded like it was aligning the sidenav contents or something

**Comment by mmalerba** [comment]
renamed to `side` to `position`

**Comment by jelbourn** [comment]
Unit tests look to be failing

**Comment by mmalerba** [comment]
fixed tests and rebased

**Comment by jelbourn** [comment]
Just needs rebease

**Comment by mmalerba** [comment]
rebased again

**Pull request merged**